packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -203,15 +203,104 @@ exports.sourceNodes = async (
       }
 
       node.internal.contentDigest = createContentDigest(node)
-      nodes.push(node)
+      nodes[node.id] = node
     })
   })
 
+  let nodeDatumsBreak = false
+
+  // Map media entities
+  for (const i in nodes) {
+    if (nodes[i]) {
+      const node = nodes[i]
+
+      // Check if the node is a file with no alt data attached
+      if (node.internal.type === `file__file` && !node.alt) {
+        // Check if a relationships property exists in the node
+        if (node.relationships) {
+          // For each relationship, check for a `node` type
+          for (const type in node.relationships) {
+            if (/^node--|^paragraph--/.test(type) && node.relationships[type]) {
```

Comment:
Because different modules name the types differently. 

For example, the Paragraph module. These are nested entities and the logic follows the same structure.